content/5_Plat/RURQ.mdx

```diff
@@ -152,135 +152,127 @@ Given the mechanics behind the `pushdown` function, all you need is a regular
 range-sum segment tree to solve the question.
 
 ```cpp
-
-#include <iostream>
-using ll = long long;
+#include <bits/stdc++.h>
 using namespace std;
+using ll = long long;
 
-const int maxN = 2e5 + 5;
-
-int N, Q;
-int a[maxN];
+enum QueryType { ADD, SET, NONE };
 
-struct node {
-	ll val;
-	ll lzAdd;
-	ll lzSet;
-	node(){};
-} tree[maxN << 2];
+struct Query {
+	QueryType type = NONE;
+	ll val = 0;
+};
 
-#define lc p << 1
-#define rc (p << 1) + 1
+/**
+ * T = tree node type, which will be long long
+ * U = update type, which will be the Query struct
+ */
+template <class T, class U> class LazySegtree {
+  private:
+	const int sz;
+	vector<T> t;
+	vector<U> lz;
```

Comment:
i guess `t` stands for `tree`, could you rename it to that
also what does `lz` actually contain
comments, variable names, whatever you think helps

Reply:
bet

Reply:
ok I added some comments - mostly with the assumption that they know how to write segtree